### ISSUE TITLE
[BUG] TGL-H IPC4 Zephyr build is broken in main

### ISSUE THREAD

**Issue by kv2019i** [thesofproject/sof#6710]
**Describe the bug**
The IPC4 native drivers is build is broken since commit "tgl: ipc4: zephyr: move to native drivers" as the IPC4 overlay for TGL-H was also updated to native drivers, but upstream Zephyr does not have all the needed definitions for intel_adsp_cavs25_tgph.

**To Reproduce**
Build SOF with
./scripts/xtensa-build-zephyr.py -i IPC4 tgl-h

**Reproduction Rate**
100%

**Expected behavior**
Firmware can be built

**Impact**
Showstopper

**Environment**
1) Branch name and commit hash of the 2 repositories: sof (firmware/topology) and linux (kernel driver).
    * Kernel:  c693d0caf6fa
    * SOF: f6f9b62d5a1
2) Name of the topology file
    * Topology: sof-hda-generic-2ch.tplg
3) Name of the platform(s) on which the bug is observed.
    * Platform: TGL-H

**Screenshots or console output**

```
/home/kvehmane/work/sof/src/ipc/ipc-zephyr.c: In function ‘ipc_platform_complete_cmd’:
/home/kvehmane/work/sof/src/ipc/ipc-zephyr.c:142: error: ‘__device_dts_ord_DT_N_NODELABEL_adsp_host_ipc_ORD’ undeclared (first use in this function)
/home/kvehmane/work/sof/src/ipc/ipc-zephyr.c:142: error: (Each undeclared identifier is reported only once
/home/kvehmane/work/sof/src/ipc/ipc-zephyr.c:142: error: for each function it appears in.)
/home/kvehmane/work/sof/src/ipc/ipc-zephyr.c: In function ‘ipc_platform_send_msg’:
/home/kvehmane/work/sof/src/ipc/ipc-zephyr.c:151: error: ‘__device_dts_ord_DT_N_NODELABEL_adsp_host_ipc_ORD’ undeclared (first use in this function)
/home/kvehmane/work/sof/src/ipc/ipc-zephyr.c: In function ‘platform_ipc_init’:
/home/kvehmane/work/sof/src/ipc/ipc-zephyr.c:175: error: ‘__device_dts_ord_DT_N_NODELABEL_adsp_host_ipc_ORD’ undeclared (first use in this function)
 ```

**Notes**

**Comment by kv2019i** [comment]
FYI @mengdonglin @lgirdwood  most probably affects sof-2.4 stable branch.

**Comment by marc-hb** [comment]
Why is this very recent tgl-h build successful?
https://sof-ci.01.org/sofpr/PR6733/build2695/build/tgl-h_zph.txt

**Comment by marc-hb** [comment]
Adding -i IPC4 tgl-h is a 4 characters addition in https://github.com/thesofproject/sof/blob/main/.github/workflows/zephyr.yml

**Comment by kv2019i** [comment]
@marc-hb wrote:
> Adding -i IPC4 tgl-h is a 4 characters addition in https://github.com/thesofproject/sof/blob/main/.github/workflows/zephyr.yml

I tried but I had to use more than four:
https://github.com/thesofproject/sof/pull/6816

**Comment by kv2019i** [comment]
Fixed with #6807 , closing.
